### PR TITLE
Make widgets easier to change appearance

### PR DESCRIPTION
**Pull request by CodexAdrian**
This PR makes the dropdown and list widget easier to manipulate and change their appearance, as well as making several private fields protected for easier use while extending

### CODE REVIEW

File: common/src/main/java/earth/terrarium/olympus/client/components/base/ListWidget.java [hunk 1]
```diff
@@ -22,17 +23,18 @@ public class ListWidget extends BaseParentWidget {
...
-    private final List<Item> items = new ArrayList<>();
+    protected final List<Item> items = new ArrayList<>();
 
-    private double scroll = 0;
-    private int lastHeight = 0;
-    private boolean scrolling = false;
+    protected double scroll = 0;
+    protected int lastHeight = 0;
+    protected boolean scrolling = false;
+    protected int spacing = 0;
```
Comment (ThatGravyBoat) [inline common/src/main/java/earth/terrarium/olympus/client/components/base/ListWidget.java]: this should be called gap

File: common/src/main/java/earth/terrarium/olympus/client/components/base/ListWidget.java [hunk 2]
```diff
@@ -75,35 +83,52 @@ public void renderWidget(GuiGraphics graphics, int mouseX, int mouseY, float par
...
+            int scrollBarY = this.getY() + getScrollbarPadding() + Math.round(((float) this.scroll / (float) this.lastHeight) * this.height);
 
-            graphics.blitSprite(
-                SCROLLBAR_THUMB,
-                scrollBarX + SCROLLBAR_PADDING,
-                scrollBarY,
-                SCROLLBAR_WIDTH,
-                scrollBarHeight
-            );
+            renderScrollbar(graphics, scrollBarX, scrollBarY, scrollBarHeight);
```
Comment (ThatGravyBoat) [inline common/src/main/java/earth/terrarium/olympus/client/components/base/ListWidget.java]: pass in mouseX, mouseY as its needed to change cursors

File: common/src/main/java/earth/terrarium/olympus/client/components/dropdown/DropdownEntry.java [hunk 3]
```diff
@@ -31,7 +33,7 @@ protected void renderWidget(GuiGraphics graphics, int mouseX, int mouseY, float
 
         Component text = this.dropdown.getText(this.value);
         int textOffset = (this.height - 8) / 2;
-        graphics.drawString(Minecraft.getInstance().font, text, this.getX() + textOffset, this.getY() + textOffset, Dropdown.COLOR);
+        graphics.drawString(Minecraft.getInstance().font, text, this.getX() + textOffset, this.getY() + textOffset, COLOR);
```
Comment (ThatGravyBoat) [inline common/src/main/java/earth/terrarium/olympus/client/components/dropdown/DropdownEntry.java]: also allow the color and textures to be overrode here too to be consistent.

File: common/src/main/java/earth/terrarium/olympus/client/components/dropdown/DropdownScreen.java [hunk 4]
```diff
@@ -35,24 +33,15 @@ public int width() {
     }
 
     public int height() {
-        return Math.min(24 * 5, this.dropdown.options().size() * 24) + 3;
+        return Math.min(24 * 5, this.dropdown.options().size() * this.dropdown.getEntryHeight()) + 3;
```
Comment (ThatGravyBoat) [inline common/src/main/java/earth/terrarium/olympus/client/components/dropdown/DropdownScreen.java]: you are missing that the height is 5 entries tall, this should always be 5 entries tall even if you change the entry height

**Review by ThatGravyBoat — CHANGES_REQUESTED** [review]

File: common/src/main/java/earth/terrarium/olympus/client/components/dropdown/DropdownEntry.java (hunk 3 above)
Comment (CodexAdrian) [reply common/src/main/java/earth/terrarium/olympus/client/components/dropdown/DropdownEntry.java]: I feel like this one is simple enough to just override. This entire widget is pretty simple to just replace in `initEntries` inside the dropdown class

Comment (ThatGravyBoat) [reply common/src/main/java/earth/terrarium/olympus/client/components/dropdown/DropdownEntry.java]: its about principles not about how easy it is, if you wanted to just change the texture you have to copy everything in this render widget and just change the 2 textures

**Review by ThatGravyBoat — APPROVED** [review]

**Pull request merged**